packages/flutter_tools/gradle/aar_init_script.gradle

```diff
@@ -0,0 +1,117 @@
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.maven.MavenDeployer
+import org.gradle.api.plugins.MavenPlugin
+import org.gradle.api.tasks.Upload
+
+void configureProject(Project project, File outputDir) {
+    if (!project.hasProperty("android")) {
+        throw new GradleException("Android property not found.")
+    }
+    if (!project.android.hasProperty("libraryVariants")) {
+        throw new GradleException("Can't generate AAR on a non Android library project.");
+    }
+
+    project.apply plugin: "maven"
+
+    project.android.libraryVariants.all { variant ->
+        addAarTask(project, variant)
+    }
+    // Snapshot versions include the timestamp in the artifact name.
+    project.version = project.version.replace("-SNAPSHOT", "")
+
+    project.uploadArchives {
+        repositories {
+            mavenDeployer {
+                repository(url: "file://${outputDir}/outputs/repo")
+            }
+        }
+    }
+    Boolean usesFlutterPlugin = project.plugins.find { it.class.name == "FlutterPlugin" } != null
+    if (!usesFlutterPlugin) {
+        // Plugins don't include their dependencies under the assumption that the parent project adds them.
```

Comment:
Are we sure these are the only libraries that plugins will be depending on?

For instance, what if plugins instruct users today to add dependencies to their own build.gradle, and now that breaks.

Reply:
Talked offline - this is needed so that gradle can use existing builds, so that rebuilds don't take more time.

Reply:
Solved offline!

Reply:
I'm not sure this one got resolved.  What are we going to do/print if we fail to build a plugin because it is missing dependencies, but it used to build because the application listed those dependencies?

Reply:
This is under a flag now, so we will see.